sdks/java/maven-archetypes/examples/build.gradle

```diff
@@ -17,10 +17,53 @@
  */
 
 apply from: project(":").file("build_rules.gradle")
-applyJavaNature()
+applyJavaNature(artifactId: "beam-sdks-java-maven-archetypes-examples")
 
 description = "Apache Beam :: SDKs :: Java :: Maven Archetypes :: Examples"
 
+// Extract the version from a library:version dependency.
+def ver = { return it.substring(1+it.lastIndexOf(':')) }
+
+processResources {
+  filter org.apache.tools.ant.filters.ReplaceTokens, tokens: [
+    'project.version':  version,
+    'bigquery.version': ver(project.library.java.google_api_services_bigquery),
+    'google-clients.version': ver(project.library.java.google_api_client),
+    'guava.version': ver(project.library.java.guava),
+    'hamcrest.version': ver(project.library.java.hamcrest_library),
+    'jackson.version': ver(project.library.java.jackson_core),
+    'joda.version': ver(project.library.java.joda_time),
+    'junit.version': ver(project.library.java.junit),
+    'pubsub.version': ver(project.library.java.google_api_services_pubsub),
+    'slf4j.version': ver(project.library.java.slf4j_api),
+    'spark.version': ver(project.library.java.spark_core),
+    'hadoop.version': ver(project.library.java.hadoop_client),
+    'mockito.version': ver(project.library.java.mockito_core),
+    'maven-compiler-plugin.version': '3.7.0',
```

Comment:
It would be useful to elevate all the Maven dependencies to the project root build.gradle file so that its not duplicated or each archetype.

Reply:
Done. Moved to root build.gradle in project.ext.library.maven